lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -4533,6 +4533,127 @@ class DecomposeAtenSignOp : public OpRewritePattern<AtenSignOp> {
 };
 } // namespace
 
+// The `embedding_dense_backward` op is decomposed as follows:
```

Comment:
I wonder if managing the decomposition here instead of relying on the upstream decomposition in dynamo is a scalable approach in the long term.

I was actually expecting us to do the opposite and move our decompositions into pytorch over time.
(Also considering that the python code for the decomposition is shorter and more readable than the corresponding MLIR C++ code).

Reply:
Hi @mgehre-amd, this comment (https://github.com/nod-ai/SHARK/pull/1567#issuecomment-1601062045) might provide more context. Please let me know if you still have any doubts or thoughts.

Reply:
> I wonder if managing the decomposition here instead of relying on the upstream decomposition in dynamo is a scalable approach in the long term.

Sorry, I missed that your query is specific to the dynamo. Yeah, we can use the upstream PyTorch decomposition it's just a choice. I can revert the dynamo changes to use the upstream decomposition instead of the Torch-MLIR one. @ramiro050 Should I do that?

Reply:
It seems the Long term roadmap (https://github.com/llvm/torch-mlir/blob/main/docs/images/roadmap_frontend.png) also mentions that we should reuse the PyTorch infra. I think for a temporary solution to unblock us this patch can be merged(as there is no harm, and there exists a lot of such instances already) and then we can take a look at fixing the issue in upstream PyTorch.

Reply:
I think fixing things upstream would require very simple changes (changing `_unsafe_index` to `index_impl` or adding type annotations). If the decomposition is broken for us, it is likely broken for other PyTorch users, so I would prefer we fix it upstream than have a workaround here.